From 7e3c17881bfdec96ee4f50ba99f29f60f7ea1b7d Mon Sep 17 00:00:00 2001
From: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
Date: Sun, 31 Jul 2016 03:49:18 +0300
Subject: [PATCH] armv8: rcar_gen3: fix config_regs table

This fixes incorrect config_regs table fill.
Without this the INOUTSEL did not reach, hence
the GPIO direction failed to set.

Signed-off-by: Vladimir Barinov <vladimir.barinov@cogentembedded.com>
---
 arch/arm/cpu/armv8/rcar_gen3/pfc-r8a7795.c | 33 +++++++++++++++++++++++++++++-
 1 file changed, 32 insertions(+), 1 deletion(-)

diff --git a/arch/arm/cpu/armv8/rcar_gen3/pfc-r8a7795.c b/arch/arm/cpu/armv8/rcar_gen3/pfc-r8a7795.c
index 456b408..766114c 100644
--- a/arch/arm/cpu/armv8/rcar_gen3/pfc-r8a7795.c
+++ b/arch/arm/cpu/armv8/rcar_gen3/pfc-r8a7795.c
@@ -4459,7 +4459,12 @@ static struct pinmux_cfg_reg pinmux_config_regs[] = {
 		}
 	},
 	{ PINMUX_CFG_REG_VAR("MOD_SEL2", 0xE6060508, 32,
-				1, 1, 1, 26, 2, 1) {
+				1, 1, 1,
+				1, 1, 1, 1, 1, 1, 1, 1,
+				1, 1, 1, 1, 1, 1, 1, 1,
+				1, 1, 1, 1, 1, 1, 1, 1,
+				1, 1,
+				2, 1) {
 		/* I2C_SEL_5 [1] */
 		FN_I2C_SEL_5_0,
 		FN_I2C_SEL_5_1,
@@ -4470,6 +4475,32 @@ static struct pinmux_cfg_reg pinmux_config_regs[] = {
 		FN_I2C_SEL_0_0,
 		FN_I2C_SEL_0_1,
 		/* reserved [26] */
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
+		0, 0,
 		/* SEL_VSP [2] */
 		FN_SEL_VSP_0,
 		FN_SEL_VSP_1,
-- 
1.9.1

